### DISCUSSION TITLE
Add "Use edge color" options to Text and Background Node Colors

### DISCUSSION THREAD

**Discussion by silopolis** [freeplane/freeplane#537] [Ideas]
Hi,

First thanks a ton for the continuous work on this essential application 👍 🙏 

Object says it all I believe. With the sooo handy level styling and automatic edge color features, the only option I'm missing are being able to have either text or background color "follow" edge color.
This would be specially handy to easily adapt a map design to light/dark backgrounds and/or screen/print target.

Thanks again and in advance for considering this request

TY
J

**Comment by macmarrum** [comment]
I understand the idea. I can see how it could be useful to you.

Please note that this can be achieved with a [script](https://docs.freeplane.org/#/scripting/Scripting). Here's one possibility: [macmarrumNCL.groovy](https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/macmarrumNCL.groovy) – `applyEdgeColorsToBranchesAndAlteringColorsToLeafs`. Let me know if you need more information.

In #520 there's a recent discussion around using NodeChangeListener (NCL) – https://github.com/freeplane/freeplane/discussions/520#discussioncomment-3308007 – and other ways of automatic execution of scripts in a mindmap – https://github.com/freeplane/freeplane/discussions/520#discussioncomment-3311709

**Comment by dpolivaev** [comment]
I think that "follows edge color" checkboxes could be added to Text and Background Node Colors and to the Cloud Color.
Community feedback wished.

**Reply by macmarrum** [reply]
I like the idea 👍 

Note: for Cloud Color, it will probably be required to split setting Color from activating Cloud. It would actually be desirable to be able to pre-set a color and later only activate Cloud.
